tests/test_integration.py

```diff
@@ -17,6 +20,57 @@
     from http.server import BaseHTTPRequestHandler, HTTPServer
 
 
+class TestHandler(BaseHTTPRequestHandler):
+    num_requests = 0
+
+    requests = defaultdict(Queue)
+
+    @classmethod
+    def _add_request(cls, req):
+        q = cls.requests[id(cls)]
+        q.put(req)
+
+    @classmethod
+    def get_requests(cls, n) -> List[BaseHTTPRequestHandler]:
+        reqs = []
+        for _ in range(n):
+            reqs.append(cls.requests[id(cls)].get(False))
+
+        assert cls.requests[id(cls)].empty()
+        return reqs
+
+    def do_GET(self):
+        return self._do_request()
+
+    def do_POST(self):
+        return self._do_request()
+
+    def _do_request(self):
+        n = self.__class__.num_requests
+        self.__class__.num_requests += 1
+        self._add_request(self)
```

Comment:
Is the handler created for each request?

Reply:
Yes. An instance of BaseHTTPRequestHandler actually represents a single incoming request.